front/lib/iam/users.ts

```diff
@@ -95,46 +89,60 @@ export async function maybeUpdateFromExternalUser(
 
 export async function createOrUpdateUser(
   session: SessionWithUser
-): Promise<{ user: UserType; created: boolean }> {
+): Promise<{ user: UserResource; created: boolean }> {
   const { user: externalUser } = session;
 
   const user = await fetchUserFromSession(session);
 
   if (user) {
+    const updateArgs: { [key: string]: string } = {};
+
     // We only update the user's email if the email is verified.
     if (externalUser.email_verified) {
-      user.email = externalUser.email;
+      updateArgs.email = externalUser.email;
     }
 
     // Update the user object from the updated session information.
-    user.username = externalUser.nickname;
-    user.name = externalUser.name;
+    updateArgs.username = externalUser.nickname;
 
     if (!user.firstName && !user.lastName) {
       if (externalUser.given_name && externalUser.family_name) {
-        user.firstName = externalUser.given_name;
-        user.lastName = externalUser.family_name;
+        updateArgs.firstName = externalUser.given_name;
+        updateArgs.lastName = externalUser.family_name;
       } else {
         const { firstName, lastName } = guessFirstAndLastNameFromFullName(
           externalUser.name
         );
-        user.firstName = firstName;
-        user.lastName = lastName;
+        updateArgs.firstName = firstName;
+        updateArgs.lastName = lastName || "";
       }
     }
 
-    await user.save();
+    if (Object.keys(updateArgs).length > 0) {
+      const needsUpdate = Object.entries(updateArgs).some(
+        ([key, value]) => user[key as keyof typeof user] !== value
```

Comment:
`nit` not a big fan, could be removed if user.updateInfo was taking an object 😉 

(can be fixed in a follow up PR)